src/NuGetGallery.Services/PackageManagement/PackageFilter.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web.Routing;
+using NuGet.Services.Entities;
+using NuGetGallery.Services.Helpers;
+
+namespace NuGetGallery.Services
+{
+    public class PackageFilter : IPackageFilter
+    {
+        private readonly IPackageService _packageService;
+
+        public PackageFilter(IPackageService packageService)
+        {
+            _packageService = packageService;
```

Comment:
Please add a null check here to catch dependency injection issues:

```suggestion
            _packageService = packageService ?? throw new ArgumentNullException(nameof(packageService));
```